nixos/modules/services/web-apps/stirling-pdf.nix

```
@@ -0,0 +1,168 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    nullOr
+    path
+    port
+    str
+    submodule
+    ;
+
+  cfg = config.services.stirling-pdf;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ thubrecht ];
+
+  options.services.stirling-pdf = {
+    enable = mkEnableOption "Stirling-PDF, a web application for manipulating PDF files";
+
+    package = mkPackageOption pkgs "stirling-pdf" { };
+
+    domain = mkOption {
+      type = str;
+      description = ''
+        The domain for the application.
+      '';
+    };
+
+    port = mkOption {
+      type = port;
+      default = 8080;
+      description = ''
+        Port that Stirling-PDF will listen on.
+      '';
+    };
+
+    environment = mkOption {
+      type = attrsOf str;
+      defaultText = ''
+        {
+          SERVER_PORT = cfg.port;
+        }
+      '';
+      description = ''
+        Settings for the stirling-pdf app.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        File containing additional environment variables to pass to Stirling PDF.
+      '';
+    };
+
+    nginx = mkOption {
+      type = nullOr (submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }));
```

```suggestion
      type = nullOr options.services.nginx.virtualHosts.type;
```

it might be better here to reference the vhost option's type via the module system (add `options` to the module's arguments), so as to not depend on relative file paths as an API (which would break if e.g. the project ever decides to do something like the pkgs/by-name migration with modules, or if the nginx module's files are otherwise restructured)